lib/deimos/consumer.rb

```diff
@@ -19,8 +19,17 @@ class Consumer
     class << self
       # @return [Deimos::SchemaBackends::Base]
       def decoder
-        @decoder ||= Deimos.schema_backend(schema: config[:schema],
-                                           namespace: config[:namespace])
+        return @decoder if @decoder
+
+        @decoder = if Utils::SchemaClass.use?(config.to_h)
+                     # Initialize an instance of the provided schema
+                     # in the event the schema class is an override, the inherited
+                     # schema and namespace will be applied
+                     schema_class = "Schemas::#{config[:schema]}".constantize.new
+                     Deimos.schema_backend(schema: schema_class.schema, namespace: schema_class.namespace)
```

Comment:
Shouldn't this be moved into the `schema_backend` method rather than just used here?